app/scripts/modules/amazon/loadBalancer/configure/createLoadBalancer.controller.js

```diff
@@ -353,13 +353,20 @@ module.exports = angular.module('spinnaker.loadBalancer.aws.create.controller',
         updateAvailableSecurityGroups(availableVpcIds);
       if (subnetPurpose) {
         $scope.loadBalancer.vpcId = availableVpcIds.length ? availableVpcIds[0] : null;
+        if (!$scope.state.hideInternalFlag && !$scope.state.internalFlagToggled) {
+          $scope.loadBalancer.isInternal = subnetPurpose.indexOf('internal') > -1;
+        }
         v2modalWizardService.includePage('Security Groups');
       } else {
         $scope.loadBalancer.vpcId = null;
         v2modalWizardService.excludePage('Security Groups');
       }
     };
 
+    this.isInternalFlagToggled = () => {
+      $scope.state.internalFlagToggled = true;
+    };
```

Comment:
Then name of this function makes me believe it should be returning a boolean, not set the value to true.